hazelcast/src/main/java/com/hazelcast/query/impl/IndexImpl.java

```diff
@@ -93,10 +100,15 @@ public void saveEntryIndex(QueryableEntry e) throws QueryException {
             converter = attributeType == null ? TypeConverters.IDENTITY_CONVERTER : attributeType.getConverter();
         }
 
-        Data key = e.getIndexKey();
-        Comparable newValue = e.getAttribute(attribute);
+        Comparable oldValue = null;
+        if (oldRecordValue != null) {
+            oldValue = QueryEntry.extractAttribute(attribute, e.getKey(), oldRecordValue, ss);
```

Comment:
Isn't it simplier to pull out a method that calls QueryEntry/sanitize? So you get something like:

oldValue=extractAttribute(attribute, e.getKey(), oldRecordValue)